### PR TITLE
Disallow save with draggers

### PR DESCRIPTION
**Pull request by def-**
Untested. @murpii Do you have a reproducer?
## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/server/save.cpp [hunk 1]
```diff
@@ -529,13 +529,15 @@ int CSaveTeam::Save(CGameContext *pGameServer, int Team, bool Dry)
 		if(pTeams->m_Core.Team(p->GetPlayer()->GetCid()) != Team)
 			continue;
 		if(m_MembersCount == j)
-			return 3;
+			return ESaveResult::CHAR_NOT_FOUND;
+		if(pGameServer->m_World.BlocksSave(p->GetPlayer()->GetCid()))
+			return ESaveResult::DRAGGER_ACTIVE;
```
Comment (Robyt3) [inline src/game/server/save.cpp:534]: The `DRAGGER_ACTIVE` name and `"Team can't be saved while a dragger is active"` message aren't consistent with the generic naming of the `BlocksSave` function. Maybe make the message more generic or let it include all potentially blocked elements, e.g. `"Team can't be saved with current world state (active draggers not allowed)"`

File: src/game/server/teams.cpp [hunk 2]
```diff
@@ -938,6 +938,12 @@ void CGameTeams::SwapTeamCharacters(CPlayer *pPrimaryPlayer, CPlayer *pTargetPla
 		return;
 	}
 
+	if(GameServer()->m_World.BlocksSave(pPrimaryPlayer->GetCid()) || GameServer()->m_World.BlocksSave(pTargetPlayer->GetCid()))
+	{
+		GameServer()->SendChatTarget(pPrimaryPlayer->GetCid(), "Can't swap while a dragger is active");
+		return;
+	}
```
[the remark this answers is not in the document]
Comment (def-) [reply src/game/server/teams.cpp]: Ok, removed. Not sure if this is affected by the cheat.

File: src/game/server/save.cpp (hunk 1 above)
Comment (def-) [reply src/game/server/save.cpp]: Done

**Comment by murpii** [comment]
I'm still able to save while a dragger affects my tee. @def- ?

**Comment by def-** [comment]
Works for me, do you have a dragger map?
screenshot_2024-08-22_00-18-50

**Comment by murpii** [comment]
> Works for me, do you have a dragger map? ![screenshot_2024-08-22_00-18-50](https://private-user

I'll shoot you a DM on discord.

**Comment by def-** [comment]
Thanks murpi, fixed

**Pull request merged**